agent/api/task/task.go

```diff
@@ -525,6 +531,20 @@ func (task *Task) initServiceConnectContainerDependencies() {
 		container.AddContainerDependency(scContainer.Name, ContainerOrderingHealthyCondition)
 		scContainer.BuildContainerDependency(container.Name, apicontainerstatus.ContainerStopped, apicontainerstatus.ContainerStopped)
 	}
+
+	// For bridge mode, we are going to have the SC container configure netns for itself as well as for all other task
+	// containers. The configuration is done when SC container transitions from RUNNING -> RESOURCES_PROVISIONED.
+	// Therefore, each application container will also wait for scContainer.RESOURCE_PROVISIONED stage to complete
+	// before starting.
+	if task.IsNetworkModeBridge() {
+		scContainer.SetSteadyStateStatusUnsafe(apicontainerstatus.ContainerResourcesProvisioned)
+		for _, container := range task.Containers {
+			if container.IsInternal() || container == scContainer {
+				continue
+			}
+			container.BuildContainerDependency(scContainer.Name, apicontainerstatus.ContainerResourcesProvisioned, apicontainerstatus.ContainerPulled)
```

Comment:
I would agree. Only thing is that it seems we have been using `apicontainerstatus.ContainerPulled` for pause container ordering, and I couldn't find much context or reason, has there ever been any.  
I guess I'll make the change here because it'll also reduce task start-up time and I really dont see the risk. But I'm going to leave the existing ordering the way they are, if that's okay with you